File: ScaleformUI_Csharp/Menus/Pause Menus/LobbyMenu/MainView.cs
using CitizenFX.Core;
using CitizenFX.Core.Native;
using ScaleformUI.Elements;
using ScaleformUI.Menu;
using ScaleformUI.Menus;
using ScaleformUI.PauseMenu;
using ScaleformUI.PauseMenus;
using ScaleformUI.PauseMenus.Elements;
using ScaleformUI.PauseMenus.Elements.Columns;
using ScaleformUI.PauseMenus.Elements.Items;
using ScaleformUI.PauseMenus.Elements.Panels;
using ScaleformUI.Scaleforms;
using static CitizenFX.Core.Native.API;

namespace ScaleformUI.LobbyMenu
{
    public delegate void LobbyMenuOpenEvent(MainView menu);
    public delegate void LobbyMenuCloseEvent(MainView menu);

    public class MainView : TabView
    {
        public event LobbyMenuOpenEvent OnLobbyMenuOpen;
        public event LobbyMenuCloseEvent OnLobbyMenuClose;
        public MinimapPanel Minimap => coronaTab.Minimap;

        public new int Index;

        public MainView(string title) : base(title)
        {
            IsCorona = true;
            coronaTab = new PlayerListTab("corona for mainview", SColor.HUD_None);
            AddTab(coronaTab);
            InstructionalButtons = new()
            {
                new InstructionalButton(Control.PhoneSelect, UIMenu._selectTextLocalized),
                new InstructionalButton(Control.PhoneCancel, UIMenu._backTextLocalized),
            };
        }
        public MainView(string title, string subtitle) : base(title, subtitle)
        {
            IsCorona = true;
            coronaTab = new PlayerListTab("corona for mainview", SColor.HUD_None);
            AddTab(coronaTab);
            InstructionalButtons = new()
            {
                new InstructionalButton(Control.PhoneSelect, UIMenu._selectTextLocalized),
                new InstructionalButton(Control.PhoneCancel, UIMenu._backTextLocalized),
            };
        }

        public MainView(string title, string subtitle, string sideTop, string sideMid, string sideBot) : base(title, subtitle, sideTop, sideMid, sideBot)
        {
            IsCorona = true;
            coronaTab = new PlayerListTab("corona for mainview", SColor.HUD_None);
            AddTab(coronaTab);
            InstructionalButtons = new()
            {
                new InstructionalButton(Control.PhoneSelect, UIMenu._selectTextLocalized),
                new InstructionalButton(Control.PhoneCancel, UIMenu._backTextLocalized),
            };
        }

        public void SelectColumn(int column) => coronaTab.SwitchColumn(column);
        public void SelectColumn(PM_COLUMNS column) => coronaTab.SwitchColumn(column);
        public void SetupLeftColumn(PM_Column column) => coronaTab.SetupLeftColumn(column);
        public void SetupCenterColumn(PM_Column column) => coronaTab.SetupCenterColumn(column);
        public void SetupRightColumn(PM_Column column) => coronaTab.SetupRightColumn(column);

        internal new void SendPauseMenuOpen()
        {
            OnLobbyMenuOpen?.Invoke(this);
        }

        internal new void SendPauseMenuClose()
        {
            OnLobbyMenuClose?.Invoke(this);
        }
    }
}

File: ScaleformUI_Lua/src/Menus/PauseMenus/PauseMenu_corona/MainView.lua
MainView = {}
MainView.__index = MainView
setmetatable(MainView, { __index = TabView })
MainView.__call = function() return "MainView" end


---@class MainView
---@field public Title string
---@field public Subtitle string
---@field public SideTop string
---@field public SideMid string
---@field public SideBot string
---@field public SettingsColumn SettingsListColumn
---@field public PlayersColumn PlayerListColumn
---@field public MissionPanel MissionDetailsPanel
---@field public InstructionalButtons InstructionalButton[]
---@field public OnLobbyMenuOpen fun(menu: MainView)
---@field public OnLobbyMenuClose fun(menu: MainView)
---@field public TemporarilyHidden boolean
---@field public controller boolean
---@field public _focus number

function MainView.New(title, subtitle, sideTop, sideMid, sideBot)
    local base = TabView.New(title, subtitle, sideTop, sideMid, sideBot)
    base.IsCorona = true
    base.coronaTab = PlayerListTab.New("corona for mainview", SColor.HUD_None)
    base.OnLobbyMenuOpen = function(menu)
    end
    base.OnLobbyMenuClose = function(menu)
    end
    local meta = setmetatable(base, MainView)
    meta:AddTab(meta.coronaTab)
    meta.Minimap = meta.coronaTab.Minimap
    meta.Minimap.Parent = meta
    return meta
end

function MainView:SelectColumn(column)
    self.coronaTab:SwitchColumn(column)
end

function MainView:SwitchColumn(column)
    self.coronaTab:SwitchColumn(column)
end

function MainView:SetupLeftColumn(column)
    self.coronaTab:SetupLeftColumn(column)
end

function MainView:SetupCenterColumn(column)
    self.coronaTab:SetupCenterColumn(column)
end

function MainView:SetupRightColumn(column)
    self.coronaTab:SetupRightColumn(column)
end